### ISSUE TITLE
$or with $regex  broken after 7.2.2 -> 7.3.0

### ISSUE THREAD

**Issue by JanGoerss** [pouchdb/pouchdb#8491]
### Issue

BugFix [0e27e23f](https://github.com/pouchdb/pouchdb/commit/0e27e23f4437bce134d0780a543f08a3020fa4de) ([#8222](https://github.com/pouchdb/pouchdb/issues/8222)) - fix multiple $regex conditions on same field when using $and

has implement error with $or and $regex

Error => userValue.every is not a function

in 
  '$regex': function (doc, userValue, parsedField, docFieldValue) {
    return fieldExists(docFieldValue) &&
      typeof docFieldValue == "string" &&
      userValue.every(function (regexValue) {
        return regexMatch(docFieldValue, regexValue);
      });
  },

with Selector

{
	"$and": [
		{
			"$or": [
				{
					"field1": {
						"$regex": "/searchtext/i"
					}
				},
				{
					"field2": {
						"$regex": "/searchtext/i"
					}
				},
				{
					"field3": {
						"$regex": "/searchtext/i"
					}
				},
				{
					"field4": {
						"$regex": "/searchtext/i"
					}
				}
			]
		},
		{
			"fieldA": {
				"$exists": true
			},
			"fieldB": {
				"$exists": true
			}
		}
	]
}

### Info
- Environment: Browser
- Platform: all
- Adapter: indexeddb
- Server: CouchDB

### Reproduce
see up

**Comment by filionf** [comment]
I have the fix ready for this. I need to write the unit tests and send the PR.

**Comment by pete-hotchkiss** [comment]
@filionf looks like there was a build error on your PR. Is this something that can be sorted - seems simple enough as it's largely a linting error

**Comment by filionf** [comment]
From what I can see, the latest build went through. What seems to be missing is a reviewer.
https://github.com/pouchdb/pouchdb/pull/8493

**Comment by maxnowack** [comment]
This is still an issue. #8493 needs to be merged.

**Comment by gwen1230** [comment]
When new version will be available ?

**Comment by gwen1230** [comment]
When this fix will be available ?

**Comment by garethbowen** [comment]
I'm not planning a release. I suggest reaching out on Slack.

**Comment by pete-hotchkiss** [comment]
@garethbowen ok. I cant seem to find details of the slack channel. Is that something you can post at all ?

**Comment by garethbowen** [comment]
@pete-hotchkiss Good point...  Join the couchdb Slack here: https://join.slack.com/t/couchdb/shared_invite/zt-fa9zim0j-H04m4o_KcLdWeOxEAcwM8g

Then there's a `#pouchdb` channel. The community is more active there than here.

The pouchdb website should also be updated to include this info...

**Comment by janl** [comment]
This has been released now, thanks for the heads-up: https://pouchdb.com/2022/11/11/pouchdb-7.3.1.html

**Comment by pete-hotchkiss** [comment]
@JanGoerss is it perhaps just a case of it not making the release notes. Personally - I haven't had a chance to run any tests yet to tell.

**Comment by filionf** [comment]
From what I can tell, this PR has not been integrated into the 7.3.x branch.
I installed a few 7.3.1 versions on my local environment and could not find the modifications that this PR brings.

**Comment by tahpot** [comment]
This is still a problem...

**Comment by thaibault** [comment]
still not working
